CodeListLibrary_project/clinicalcode/entity_utils/constants.py

```diff
@@ -0,0 +1,242 @@
+'''
+PAGE_RESULTS_SIZE = {
+    '1': 20,
+    '2': 50,
+    '3': 100
+}
+
+'''
+    Used for:
+        - Hashmap is used to determine how to get values from sourced data e.g. tags, collections
+          that are accounted for in the metadata portion of an entity
+
+        - By filter to determine how and when to render metadata-related filters
+'''
+metadata = {
+    'template': {
+        'title': 'Entity Type',
+        'active': True,
+        'field_type': 'int',
+        'source': 'Template',
+        'query': 'id',
+        'relative': 'name',
+        'computed': True,
+        'filterable': True,
+        'single_search_only': True,
+        "validation": {
+            "mandatory": True
+        }
+    },
+    "name": {
+        "title": "Name",
+        "active": True,
+        "field_type": "string",
+        "desired_input": "inputbox",
+        "validation": {
+            "mandatory": True
+        }
+    },
+    "definition": {
+        "title": "Definition",
+        "active": True,
+        "field_type": "string",
+        "desired_input": "markdown",
+        "desired_output": "markdown",
+        "validation": {
+            "mandatory": False
+        }
+    },
+    "implementation": {
+        "title": "Implementation",
+        "active": True,
+        "field_type": "string",
+        "desired_input": "markdown",
+        "desired_output": "markdown",
+        "validation": {
+            "mandatory": False
+        }
+    },
+    "publications": {
+        "title": "Publications",
+        "active": True,
+        "field_type": "string_array",
+        "desired_input": "inputbox",
+        "validation": {
+            "mandatory": False
+        }
+    },
+    "validation": {
+        "title": "Validation",
+        "active": True,
+        "field_type": "string",
+        "desired_input": "markdown",
+        "desired_output": "markdown",
+        "hide_if_empty": True,
+        "validation": {
+            "mandatory": False
+        }
+    },
+    "citation_requirements": {
+        "title": "Citation Requirements",
+        "active": True,
+        "field_type": "textarea",
+        "desired_input": "inputbox",
+        "hide_if_empty": True,
+        "validation": {
+            "mandatory": False
+        }
+    },
+    'created': {
+        'title': 'Date',
+        'active': True,
+        'field_type': 'datetime',
+        'computed': True,
+        'filterable': True,
+        "validation": {
+            "mandatory": True
+        }
+    },
+    'author': {
+        'title': 'Author',
+        'active': True,
+        'field_type': 'string',
+        'desired_input': 'inputbox',
+        'searchable': True,
+        "validation": {
+            "mandatory": True
+        }
+    },
+    'collections': {
+        'title': 'Collections',
+        'active': True,
+        'field_type': 'int_array',
+        'source': 'Tag',
+        'query': 'id',
+        'relative': 'description',
+        'filter': {
```

Comment:
We need to look into cleaning up tags and collections, as well as other data models - we use different nomenclature for PK and the name field which increases complexity of querying these fields